boa/src/builtins/string/mod.rs

```diff
@@ -50,11 +50,11 @@ pub(crate) fn code_point_at(string: RcString, position: i32) -> Option<(u32, u8,
 }
 
 fn is_leading_surrogate(value: u16) -> bool {
-    value >= 0xD800 && value <= 0xDBFF
+    (0xD800..0xDBFF).contains(&value)
 }
 
 fn is_trailing_surrogate(value: u16) -> bool {
-    value >= 0xDC00 && value <= 0xDFFF
+    (0xDC00..0xDFFF).contains(&value)
 }
```

Comment:
Were these the Clippy lints you mention?

Reply:
@RageKnify Indeed: https://rust-lang.github.io/rust-clippy/master/index.html#manual_range_contains

Reply:
Oh, but I think you reduced the ranges, they included `0xDBFF` and `0xDFFF` before and now they only go up to the numbers before those. I think you should change it to use `=`, for example: `(0xD800..=0xDBFF)`, right?

Reply:
You're right. I misread the warning. (New to Rust, and clippy for that matter.)

Reply:
@RageKnify Fixed and added tests around that.